### ISSUE TITLE
Images, Containers and Volumes gone after update

### ISSUE THREAD

**Issue by Geosynopsis** [docker/for-win#2498]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 6F30ECE2-F064-46DC-9CA0-E33D6D1EA884/20180828105054

  - Windows Version: Windows Server 16
  - Docker for Windows Version:  18.06.0-ce-win72

Hi, 
After the recent update to 18.06.0, all the images, containers and the volumes from the host are deleted. The MobyLinuxHD, however, still occupies the similar amount of disk space as before the update. Has anybody encountered similar issue ? If yes, is there any way to recover at least the volumes?

Additional Information (Don't know whether it is relevant):
Before the update, the host memory was being consumed to almost full by vpnkit.exe. After the update, I got the following errors in the log:
```
[10:42:12.065][VpnKit         ][Error  ] vpnkit.exe: While watching /etc/resolv.conf: ENOENT
...

[13:00:30.290][Moby           ][Info   ] 2018-08-28T11:00:30Z,onboot.005-extend;panic: runtime error: index out of range
...

[13:00:37.837][Moby           ][Info   ] [34mINFO[0m[0000] starting containerd                           [34mrevision[0m=d64c661f1d51c48782c9cec8fda7604785f93587 [34mversion[0m=v1.1.1
[13:00:37.853][Moby           ][Info   ] [34mINFO[0m[0000] loading plugin "io.containerd.content.v1.content"...  [34mtype[0m=io.containerd.content.v1
[13:00:37.869][Moby           ][Info   ] [34mINFO[0m[0000] loading plugin "io.containerd.snapshotter.v1.btrfs"...  [34mtype[0m=io.containerd.snapshotter.v1
[13:00:37.869][Moby           ][Info   ] [33mWARN[0m[0000] failed to load plugin io.containerd.snapshotter.v1.btrfs  [33merror[0m="path /var/lib/containerd/io.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter"
[13:00:37.884][Moby           ][Info   ] [34mINFO[0m[0000] loading plugin "io.containerd.snapshotter.v1.aufs"...  [34mtype[0m=io.containerd.snapshotter.v1
[13:00:37.900][Moby           ][Info   ] [33mWARN[0m[0000] failed to load plugin io.containerd.snapshotter.v1.aufs  [33merror[0m="modprobe aufs failed: "modprobe: module aufs not found in modules.dep\n": exit status 1"
[13:00:37.900][Moby           ][Info   ] [34mINFO[0m[0000] loading plugin "io.containerd.snapshotter.v1.native"...  [34mtype[0m=io.containerd.snapshotter.v1
[13:00:37.915][Moby           ][Info   ] [34mINFO[0m[0000] loading plugin "io.containerd.snapshotter.v1.overlayfs"...  [34mtype[0m=io.containerd.snapshotter.v1
[13:00:37.931][Moby           ][Info   ] [34mINFO[0m[0000] loading plugin "io.containerd.snapshotter.v1.zfs"...  [34mtype[0m=io.containerd.snapshotter.v1
[13:00:37.947][Moby           ][Info   ] [33mWARN[0m[0000] failed to load plugin io.containerd.snapshotter.v1.zfs  [33merror[0m="path /var/lib/containerd/io.containerd.snapshotter.v1.zfs must be a zfs filesystem to be used with the zfs snapshotter"
[13:00:37.962][Moby           ][Info   ] [34mINFO[0m[0000] loading plugin "io.containerd.metadata.v1.bolt"...  [34mtype[0m=io.containerd.metadata.v1
[13:00:37.978][Moby           ][Info   ] [33mWARN[0m[0000] could not use snapshotter btrfs in metadata plugin  [33merror[0m="path /var/lib/containerd/io.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter"
[13:00:37.994][Moby           ][Info   ] [33mWARN[0m[0000] could not use snapshotter aufs in metadata plugin  [33merror[0m="modprobe aufs failed: "modprobe: module aufs not found in modules.dep\n": exit status 1"
[13:00:38.009][Moby           ][Info   ] [33mWARN[0m[0000] could not use snapshotter zfs in metadata plugin  [33merror[0m="path /var/lib/containerd/io.containerd.snapshotter.v1.zfs must be a zfs filesystem to be used with the zfs snapshotter"

...

[13:00:39.775][Moby           ][Info   ] [31mERRO[0m[0002] failed to create task                         [31merror[0m="OCI runtime create failed: container_linux.go:348: starting container process caused "process_linux.go:402: container init caused \"rootfs_linux.go:58: mounting \\\"/run/config/socks\\\" to rootfs \\\"/containers/services/socks/rootfs\\\" at \\\"/run/config/socks\\\" caused \\\"stat /run/config/socks: no such file or directory\\\"\"": unknown"

```

EDIT :
One more thing I noticed is that everytime I restart the docker, all images, volumes, and containers gone. I suspect that while restarting the docker isn't able to access MobyLinuxVM.vhdx so it creates some temporary drive. I can't find where those temporary drives are created. I am investigating more. Will keep updating

**Comment by Geosynopsis** [comment]
I updated the docker from 18.06.0-ce-win72 to 18.06.1-ce-win73 but there wasn't any change in the situation. I was still not able to get access to anything on VHDX. 

I had an older version of DoW (18.03.1-ce-win65) running on another server and I updated it to 18.06.1-ce-win73 it turned out to be fine. So I reinstalled the docker on the server with the problem and recovered the volumes from a copy of old VHDX. Now it works like charm. 

**What I think was a problem:**
When the update for 18.06.0 was rolled, the samba permissions were changed "to avoid issue when trying to manipulate files with different users in containers". So it might have rendered the docker to not have permissions over some files essential for mounting old VHDX. But on the update for 18.06.1, the samba permissions were changed back to default. Therefore, there was no effect when updating from 18.03.1 to 18.06.1.
I am not 100 percent sure whether it is the case. It would be nice to have some more information on this regard.

But for now, I guess I can close the issue.
